sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Internal/WebPubSubServiceClientFactory.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.Options;
+using System;
+
+namespace Microsoft.Azure.WebPubSub.AspNetCore
+{
+    internal class WebPubSubServiceClientFactory
+    {
+        private readonly WebPubSubOptions _options;
+
+        public WebPubSubServiceClientFactory(IOptions<WebPubSubOptions> options)
+        {
+            _options = options.Value;
+        }
+
+        public WebPubSubServiceClient<THub> Create<THub>() where THub : WebPubSubHub
+        {
+            if (_options == null || _options.ServiceEndpoint == null)
+            {
+                throw new ArgumentException($"Not able to create the WebPubSubServiceClient without a valid WebPubSubOptions.ServiceEndpoint. Please configure the ServiceEndpoint when DI the WebPubSub.");
+            }
+
+            if (!string.IsNullOrEmpty(_options.ServiceEndpoint.ConnectionString))
```

Comment:
I am a little worried if later we support more properties in `ServiceEndpoint`. Instead of such implicit fallback, how about using some `kind` enum so that we can throw if new .ctors are not supported?